src/otx/algo/callbacks/tile_sampling_hook.py

```diff
@@ -0,0 +1,25 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Tile Sampling Hook for OTXTileTrainDataset."""
+from lightning import Callback, LightningModule, Trainer
+
+
+class TileSamplingHook(Callback):
```

Comment:
Why do we need this?

Reply:
@vinnamkim This PR mainly brings the tiling features from OTX 1.x to here.

Tile sampling is employed to expedite training duration by selecting a portion of samples. Through extensive experimentation, we observed the efficacy (faster training while maintaining good accuracy) of tile sampling particularly for Geti dataset (Vitens). However, numerous tiles can significantly impede training speed. This enhancement primarily targets Geti users who prioritize prompt results. Tiling processes, both in training and inference, are time-consuming. We (Jauguk, Songki, Sungman and I) dedicated 2-3 weeks to optimizing tiling in OTX 1.x. For further insights, you could DM me or refer to the folks involved. I don't think it's ideal to elaborate on all previous activities in this PR.

Reply:
I still don't know why it is needed. But if so, it seems better to add [RandomSampler](https://pytorch.org/docs/stable/data.html#torch.utils.data.RandomSampler) with `num_samples` argument to `torch.DataLoader` at initializing `OTXDataModule` rather than adding this callback. The current way can create a hidden constraint for the configurations (sampling prob < 1.0 should include this callback).

Reply:
Thank you for the advice. I've implemented your suggestion by adding `RandomSampler` to train dataloader, which is a better approach. Also, I've removed the Datumaro sampling and tile sampling hook and made other necessary adjustments.